diff --git a/sourcekit-lsp/Utilities/build-script-helper.py b/sourcekit-lsp/Utilities/build-script-helper.py
index 17ad9e3..953fffd 100755
--- a/sourcekit-lsp/Utilities/build-script-helper.py
+++ b/sourcekit-lsp/Utilities/build-script-helper.py
@@ -142,6 +142,7 @@ def get_swiftpm_options(swift_exec: str, args: argparse.Namespace) -> List[str]:
         elif re.match('android-', args.cross_compile_host):
             print('Cross-compiling for %s' % args.cross_compile_host)
             swiftpm_args += ['--destination', args.cross_compile_config]
+            swiftpm_args += ['--pkg-config-path', os.path.join(os.path.split(args.cross_compile_config)[0], 'lib/pkgconfig')]
         else:
             fatal_error("cannot cross-compile for %s" % args.cross_compile_host)
 
diff --git a/swiftpm/Package.swift b/swiftpm/Package.swift
index 48cd98431..6eaacae00 100644
--- a/swiftpm/Package.swift
+++ b/swiftpm/Package.swift
@@ -76,7 +76,7 @@ let swiftPMProduct = (
 let systemSQLitePkgConfig: String? = nil
 #else
 let includeDynamicLibrary: Bool = true
-let systemSQLitePkgConfig: String? = "sqlite3"
+let systemSQLitePkgConfig: String? = nil
 #endif

 /** An array of products which have two versions listed: one dynamically linked, the other with the
@@ -199,7 +203,7 @@ let package = Package(
             dependencies: [
                 "_AsyncFileSystem",
                 .target(name: "SPMSQLite3", condition: .when(platforms: [.macOS, .iOS, .tvOS, .watchOS, .visionOS, .macCatalyst, .linux])),
-                .product(name: "SwiftToolchainCSQLite", package: "swift-toolchain-sqlite", condition: .when(platforms: [.windows])),
+                .product(name: "SwiftToolchainCSQLite", package: "swift-toolchain-sqlite", condition: .when(platforms: [.windows, .android])),
                 .product(name: "DequeModule", package: "swift-collections"),
                 .product(name: "OrderedCollections", package: "swift-collections"),
                 .product(name: "SwiftToolsSupport-auto", package: "swift-tools-support-core"),
diff --git a/swiftpm/Sources/Basics/SQLite.swift b/swiftpm/Sources/Basics/SQLite.swift
index 2ad235030..803490b48 100644
--- a/swiftpm/Sources/Basics/SQLite.swift
+++ b/swiftpm/Sources/Basics/SQLite.swift
@@ -12,7 +12,7 @@
 
 import Foundation
 
-#if SWIFT_PACKAGE && os(Windows)
+#if SWIFT_PACKAGE && (os(Windows) || os(Android))
 #if USE_IMPL_ONLY_IMPORTS
 @_implementationOnly import SwiftToolchainCSQLite
 #else
